src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/InternalRecordWriter.java

```diff
@@ -45,12 +51,21 @@ public InternalRecordWriter(TSFileConfig conf, TSFileIOWriter tsfileWriter, Writ
         this.writeSupport = writeSupport;
         this.schema = schema;
         this.primaryRowGroupSize = conf.groupSizeInByte;
+        this.pageSize = conf.pageSizeInByte;
         this.oneRowMaxSize = schema.getCurrentRowMaxSize();
+        assert primaryRowGroupSize > oneRowMaxSize;
         this.rowGroupSizeThreshold = primaryRowGroupSize - oneRowMaxSize;
-        this.pageSize = conf.pageSizeInByte;
         writeSupport.init(groupWriters);
     }
 
+    public void addMeasurementByJson(JSONObject measurement) throws IOException {
+        JsonConverter.addJsonToMeasurement(measurement, schema);
+        this.oneRowMaxSize = schema.getCurrentRowMaxSize();
+        assert primaryRowGroupSize > oneRowMaxSize;
```

Comment:
There will  be an error when the user add time series using iotdb, if the row max size is big the primary row size

Reply:
I've convert assert to throwing exception. This error can't be solved by file-level.